fs2-google-pubsub-grpc/src/main/scala/com/permutive/pubsub/consumer/grpc/PubsubGoogleConsumer.scala

```diff
@@ -41,8 +42,9 @@ object PubsubGoogleConsumer {
       .subscribe(projectId, subscription, config)
       .flatMap { case internal.Model.Record(msg, ack, nack) =>
         MessageDecoder[A].decode(msg.getData.toByteArray) match {
-          case Left(e)  => Stream.exec(errorHandler(msg, e, ack, nack))
-          case Right(v) => Stream.emit(ConsumerRecord(v, ack, nack, _ => Applicative[F].unit))
+          case Left(e) => Stream.exec(errorHandler(msg, e, ack, nack))
+          case Right(v) =>
+            Stream.emit(ConsumerRecord(v, msg.getAttributesMap.asScala.toMap, ack, nack, _ => Applicative[F].unit))
```

Comment:
I've had a poke around and this `getAttributesMap` should never return `null`, so no need to be defensive here. (Unfortunately we don't have gRPC tests as when this repo was first written the emulator didn't support gRPC requests...)